src/main/java/PHambot/storage/UserData.java

```diff
@@ -0,0 +1,127 @@
+import PHambot.task.ToDo;
+
+/**
+ * Handles the storage and retrieval of user data, specifically tasks lists.
+ * This class is responsible for saving tasks lists to a file and loading tasks lists from a file.
+ */
+public class UserData {
+
+    private TaskList tasks;
+    private String defaultFileDirectory = "data/testSaveFile.txt";
+    private String saveFileDirectory;
+
+    /**
+     * Constructs a UserData object with a specified directory.
+     *
+     * @param directory the directory where tasks will be saved and loaded from.
+     */
+    public UserData(String directory) {
+        saveFileDirectory = directory;
+        this.tasks = loadTasks(directory);
+    }
+
+    /**
+     * Constructs a UserData object with the default directory.
+     */
+    public UserData() {
+        saveFileDirectory = defaultFileDirectory;
+        this.tasks = loadTasks(saveFileDirectory);
+    }
+
+    public TaskList getTasks() {
+        return tasks;
+    }
+
+    public void setTasks(TaskList tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Saves the list of tasks to the instance's save file directory.
+     */
+    public void saveTasks() {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(saveFileDirectory))) {
+            for (int i = 0; i < tasks.taskCount(); i++) {
+                Task task = tasks.getTask(i);
+                if (task instanceof Deadline) {
+                    writer.write("D" + '/' + task.getTaskStatus() + "/" + task.getTaskName() + "/" + ((Deadline) task).getDate().toString());
+                    if (((Deadline) task).getTime() != null) {
+                        writer.write("/" + ((Deadline) task).getTime().toString());
+                    }
+                }
+                else if (task instanceof Event) {
+                    writer.write("E" +"/" + task.getTaskStatus() + "/" +task.getTaskName() + "/" + ((Event) task).getDate());
+                    if (((Event) task).getTime() != null) {
+                        writer.write("/" + ((Event) task).getTime().toString());
+                    }
+                }
+                else {
+                    writer.write("T" + "/" + task.getTaskStatus() + "/" + task.getTaskName());
+                }
+                writer.newLine();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Loads the list of tasks from the specified file directory.
+     *
+     * @param fileName the name of the file to load tasks from.
+     * @return the list of tasks loaded from the file.
+     */
+    private TaskList loadTasks(String fileName) {
+        TaskList taskList = new TaskList();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            String task;
+            while ((task = reader.readLine()) != null) {
+                String[] splitTask = task.split("/");
+                if (splitTask[0].equals("T")) {
```

Comment:
Would be more logically accurate if you used "else if" to run through the situations here